### PR TITLE
Adds documentation for non-AWS deployment

### PR DESCRIPTION
**Pull request by bbranan**

### CODE REVIEW

**Comment by bbranan** [comment]
Most of the information added in this PR came from: https://spotdocs.scholarsportal.info/display/DURA/Using+Duracloud+with+non-AWS+Options

File: non-aws-deployment.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+If you are upgrading from a previous version of DuraCloud, ensure that you have run the schema update script `resources/sql/update-non-aws.sql` in the Management Console Baseline. This will add required columns to the `duracloud_mill` and `global_properties` tables in the accounts database. The exact command will vary depending on your MySQL installation.
+```
+mysql <accounts_db_name> < resources/sql/update-non-aws.sql
+```
+## Email
+SMTP can be used in place of AWS SES for sending emails. Setting up an SMTP server is beyond the scope of this document, but any standard SMTP setup should work fine, with or without authentication.
+
+To start using SMTP to send emails, add the following to the `.properties` files used to configure each DuraCloud application:
+```
+notification.type=smtp
```
Comment (fozboz) [inline non-aws-deployment.md]: Now `emailer.type`
```suggestion
emailer.type=smtp
```

File: non-aws-deployment.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+notification.user=smtp_username
+notification.pass=smtp_password
+```
+
+## Queues
+RabbitMQ can be used in place of SQS to provide system queues. This was developed and tested using RabbitMQ version 3.7.17, Erlang 22.0.7. The `rabbitmq_message_timestamp` plugin must be installed. SQS functions are provided by a RabbitMQ `direct` exchange
+
+### Replacing SQS
+1. In RabbitMQ, create an Exchange of type `direct`
+1. In RabbitMQ server, create the following queues (the actual names are arbitrary, but make sure they are unique):
```
Comment (fozboz) [inline non-aws-deployment.md]:
```suggestion
2. In RabbitMQ server, create the following queues (the actual names are arbitrary, but make sure they are unique):
```

File: non-aws-deployment.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  * audit
+  * bit-integrity
+  * dup-high-priority
+  * dup-low-priority
+  * bit-report
+  * dead-letter
+  * storagestats
+1. Bind all queues created in step 2 to the exchange created in step 1, use the queue names as the routing key
```
Comment (fozboz) [inline non-aws-deployment.md]:
```suggestion
3. Bind all queues created in step 2 to the exchange created in step 1, use the queue names as the routing key
```

File: non-aws-deployment.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+1. In Management Console > root console > Duracloud MiIl:
```
Comment (fozboz) [inline non-aws-deployment.md]:
```suggestion
4. In Management Console > root console > Duracloud MiIl:
```

File: non-aws-deployment.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  * Change Queue Type to RabbitMQ
+  * Input all the config, including the direct exchange name created in step 1 and audit queue name created in step 2.
+  * Save settings and restart Tomcat
+
+## Notifications
+RabbitMQ can be used in place of SNS to provide system notifications. SNS functions are provided by a RabbitMQ `fanout` exchange.
+
+### Replacing SNS
+1. In RabbitMQ server, create an Exchange (type `fanout`)
+1. In Management Console > root console > Global Properties:
```
Comment (fozboz) [inline non-aws-deployment.md]:
```suggestion
2. In Management Console > root console > Global Properties:
```

File: non-aws-deployment.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+```
+Ensure that all of the `queue.name.*` parameters match the names of the queues you created in RabbitMQ.
+
+## OpenStack Swift Storage
+OpenStack Swift can be used as a DuraCloud Storage Provider, regardless of whether other components in the system use non-AWS options. These configurations are managed in the DuraCloud Management Console along with all other Storage Provider types.
+
+To use Swift as a DuraCloud Storage provider, you must:
+* be running OpenStack Swift Train or newer
+* install the S3 middleware API
+* create EC2 credentials for the domain/project where you want to store audit logs/duplication policies
```
Comment (fozboz) [inline non-aws-deployment.md]: I think we're specifically referring to Swift as a storage provider here, not for backend storage.
```suggestion
* create EC2 credentials for the domain/project where you want to store containers
```

File: non-aws-deployment.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+### Swift for System Storage
+OpenStack Swift can also be used in place of Amazon S3 for background DuraCloud system storage.
```
Comment (fozboz) [inline non-aws-deployment.md]:
```suggestion
OpenStack Swift can also be used in place of Amazon S3 for background DuraCloud system storage (i.e. duplication policies and audit logs).
```

File: non-aws-deployment.md (hunk 6 above)
Comment (bbranan) [reply non-aws-deployment.md]: Agreed, and adding a bit of clarification around the `containers` term
```suggestion
* create EC2 credentials for the domain/project where you want to create Swift containers to serve as DuraCloud spaces
```

**Pull request merged**